src/communication/hpxreceiver.h

```diff
@@ -79,11 +79,11 @@ class HPXReceiver : public hpx::components::simple_component_base<HPXReceiver<CA
     virtual ~HPXReceiver()
     {}
 
-    void receive(std::size_t step, Cargo&& val)
+    void receive(std::size_t step, Cargo val)
     {
-        buffer.store_received(step, std::forward<Cargo>(val));
+        buffer.store_received(step, std::move(val));
```

Comment:
The old code was not 100% correct. Cargo was an rvalue reference, so you could only move anyway. Actions are always called with rvalue references, so it doesn't really matter if the function takes the parameter by value or reference, which the small problem, that taking it by reference, didn't compile.